tests/test_word_colors.py

```diff
@@ -0,0 +1,146 @@
+                ):
+                    assert content.textbox.ncolor is None
+
+                    for word in content.textbox.words:
+                        if word.text == "has":
+                            assert word.ncolor == (0, 0, 1)
+                        elif word.text == "color":
+                            assert word.ncolor in [(0, 1, 0), (0, 0, 0)]
+                        elif word.text == "changes":
+                            assert word.ncolor == (1, 0, 0)
+                        elif word.text == "words":
+                            assert word.ncolor == (0, 0, 1)
+
+
+def test_colors_4() -> None:
+    """Test word colors in given chapter paragraph."""
+    objects = libpdf.load(PDF_COLOR_STYLE)
+    assert objects is not None
+    assert objects.flattened.chapters
+
+    for chapter in objects.flattened.chapters:
+        if "Words" in chapter.title:
+            for content in chapter.content:
+                if "This words have no color" in content.textbox.text:
+                    assert content.textbox.ncolor is None
+
+                    for word in content.textbox.words:
+                        assert word.ncolor is None or word.ncolor == (0, 0, 0)
+
+
+def test_colors_5() -> None:
+    """Test word colors in given chapter paragraph."""
+    objects = libpdf.load(PDF_COLOR_STYLE)
+    assert objects is not None
+    assert objects.flattened.chapters
+
+    for chapter in objects.flattened.chapters:
+        if "Words" in chapter.title:
+            for content in chapter.content:
+                if "These words are printed" in content.textbox.text:
+                    assert content.textbox.ncolor is None
+
+                    for word in content.textbox.words:
+                        if word.text in ["words", "but"]:
+                            assert word.ncolor == (0, 1, 0)
+                        elif word.text == "printed":
+                            assert word.ncolor == (0, 0, 1)
+                        elif word.text == "background":
+                            assert word.ncolor == (1, 0, 0)
+
+
+def test_colors_6() -> None:
+    """Test word colors in given chapter paragraph."""
+    objects = libpdf.load(PDF_COLOR_STYLE)
+    assert objects is not None
+    assert objects.flattened.chapters
+
+    for chapter in objects.flattened.chapters:
+        if "Styled Text" in chapter.title:
+            for content in chapter.content:
+                if "bold text format" in content.textbox.text:
+                    assert content.textbox.bold is None
+                    for word in content.textbox.words:
+                        if word.text == "bold":
+                            assert word.bold is True
+                        else:
+                            assert word.bold is False
+                elif "italc text format" in content.textbox.text:
+                    assert content.textbox.italic is None
+                    if word.text == "italc":
```

Comment:
```suggestion
                    if word.text == "italic":
```

Reply:
Ok, your are the pdf expert :smile: . Exposing the fontname should follow the same strategy like for ncolor? store fontname on each char, return fontname for word if all chars in the word have same font and so on.

Reply:
Haven't checked that, but I guess the `italc` typo is also in the PDF, right? If we remove the italic handling, this test code is to be removed anyway.

Reply:
Ups. fixed the typo, but not closed this conversation. But was good, line 138 also contained the typo :smile: